### ISSUE TITLE
[Leaving IBM]: QA: Screen reader focus is not restricted within the dialog

### ISSUE THREAD

**Issue by gopinathan-n** [carbon-design-system/carbon-for-ibm-dotcom#9862]
### Description

Page Element:
SR focus not trapped within the dialog.

Issue Description:
In Vo iPhone Safari, Screen reader focus is not trapped within the dialog in Left/Right swipe navigation. However in JAWS FF, initially the focus is trapped but when we navigate with the arrow keys the focus moves out of the dialog.

Recommend Fix:
Screen reader focus should be trapped within the dialog

User Impact:
When the screen reader focus is not restricted within the modal dialog, it also identifies parent page content that disorients screen reader users.

Test Tool:
JAWS FF
Voiceover safari iPhone

### Component(s) impacted

Leaving IBM
(web component canary and web component staging)

### Browser

Safari, Firefox

### Carbon for IBM.com version

v1.40.0-rc.0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

Web component Canary https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components Web component staging https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/web-components

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
